provider/cluster/manager.go

```diff
@@ -94,6 +94,7 @@ func (dm *deploymentManager) teardown() error {
 
 func (dm *deploymentManager) run() {
 	defer dm.lc.ShutdownCompleted()
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The thing with this one is that it's not clear that we should actually cancel operations when things start to shut down.

For instance if it's in the middle of closing a lease and we bounce the provider, we don't really want to cancel the api call to delete the namespace.

Reply:
Yeah, I agree we'd rather wait on that to finish (even if it takes a while). I'll go back over this and try and make sure I'm not messing anything up like that.